crates/accelerate/src/linear_matrix.rs

```diff
@@ -0,0 +1,102 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use ndarray::{s, ArrayViewMut2, Axis};
+use numpy::PyReadwriteArray2;
+use pyo3::prelude::*;
+
+// Perform ROW operation on a matrix mat
+fn _row_op(mat: &mut ArrayViewMut2<bool>, ctrl: usize, trgt: usize) {
+    let row0 = mat.row(ctrl).to_owned();
+    let mut row1 = mat.row_mut(trgt);
+    row1.zip_mut_with(&row0, |x, &y| *x ^= y);
+}
+
+// Perform COL operation on a matrix mat
+fn _col_op(mat: &mut ArrayViewMut2<bool>, ctrl: usize, trgt: usize) {
+    let col0 = mat.column(ctrl).to_owned();
+    let mut col1 = mat.column_mut(trgt);
+    col1.zip_mut_with(&col0, |x, &y| *x ^= y);
+}
+
+// Gauss elimination of a matrix mat with m rows and n columns.
```

Comment:
As a docstring, should this have 3 backslashes? Is a non-full elimination only the lower triangular elimination? If yes it might be nice to mention it explicitly

Reply:
Indeed, it's still a comment. I'll add docstrings when I'm done.